src/write_hdf5_xml/write_hdf5_xml.c

```diff
@@ -2256,7 +2272,10 @@ int main(int argc, char **argv)
     }
     sprintf(filename, "%s%s%s%s", hh, hv, vh, vv);
     fprintf(fp, "    <terrain_corrected_image>\n");
-    meta = meta_read(params->rtc_HH_metadata);
+    if (params->rtc_HH_metadata)
+      meta = meta_read(params->rtc_HH_metadata);
+    else if (params->rtc_VV_metadata)
+      meta = meta_read(params->rtc_VV_metadata);
     fprintf(fp, "      <file type=\"string\" definition=\"file name(s) of the "
```

Comment:
Maybe add this:
else {
  asfPrintError("No metadata file found!");
  meta = NULL;
}